pkg/auth/api/user/user_store.go

```diff
@@ -194,6 +204,19 @@ func (s *userStore) Update(apiContext *types.APIContext, schema *types.Schema, d
 		return nil, httperror.NewAPIError(httperror.InvalidAction, "You cannot deactivate yourself")
 	}
 
+	isAdminResource, err := s.isAdminResource(id)
+	if err != nil {
+		return nil, err
+	}
+	if isAdminResource {
+		isRestrictedAdmin, err := s.isRestrictedAdmin(apiContext)
+		if err != nil {
+			return nil, err
+		}
+		if isRestrictedAdmin {
+			return nil, httperror.NewAPIError(httperror.InvalidAction, "You cannot edit this user")
```

Comment:
nit: should this be lowercase?

```suggestion
			return nil, httperror.NewAPIError(httperror.InvalidAction, "you cannot edit this user")
```

Same for Delete